web3/main.py

```diff
@@ -89,9 +89,11 @@ class Web3(object):
 
     # Encoding and Decoding
     toHex = staticmethod(to_hex)
-    toAscii = staticmethod(decode_hex)
+    toBytes = staticmethod(decode_hex)
```

Comment:
Something to think about.  

The `force_text` and `force_bytes` functions from `eth_utils` are *wrong* in certain ways (since they use `latin1` instead of `utf8`).  I can't recall if we've had this discussion but, this came about because you can't encode all of the 256 possible bytes using utf8.

Both the `decode_hex` and `encode_hex` functions use these `force_X` functions under the hood.  

If continuing to use these functions in new apis we are increasing the number of places that these are used and they will almost definitely need to be fixed at some point in the near-to-medium-term future.

This has been on my mind for a while.  I don't know what the right fix is for this, but I'm inclined to at least start talking about it because `eth_utils` uses those functions **everywhere**.

Reply:
> I can't recall if we've had this discussion but, this came about because you can't encode all of the 256 possible bytes using utf8.

Yeah, we did. If it's an arbitrary bytestring, we shouldn't try to encode it, it should just stay `bytes` (or second-best, a hexstr). Staying as `bytes` won't be possible until py3 is required, of course.

> Both the decode_hex and encode_hex functions use these force_X functions under the hood... If continuing to use these functions in new apis we are increasing the number of places that these are used

Ok, you convinced me. I'll do this the less-lazy way. The new `toBytes` and `fromBytes` won't have exactly the same functionality then.

Between this, and `sha3`, it's about time to start including a py2 deprecation warning when initializing Web3.